test/signing_test.go

```diff
@@ -8,15 +8,420 @@ import (
+
+			runTest(t, distroSigningTest(t, spec, testConfig.SignTarget))
+		})
+
+		t.Run("target config takes precedence when root config is there", func(t *testing.T) {
+			t.Parallel()
+			spec := newSpec()
+
+			first, _, _ := strings.Cut(testConfig.SignTarget, "/")
+			spec.Targets = map[string]dalec.Target{
+				first: {
+					PackageConfig: &dalec.PackageConfig{
+						Signer: &dalec.PackageSigner{
+							Frontend: &dalec.Frontend{
+								Image: spec.PackageConfig.Signer.Image,
+							},
+						},
+					},
+				},
+			}
+
+			spec.PackageConfig.Signer.Image = "notexist"
+			runTest(t, distroSigningTest(t, spec, testConfig.SignTarget))
+		})
+
+		t.Run("with args", func(t *testing.T) {
+			t.Parallel()
+
+			spec := newSpec()
+			spec.PackageConfig.Signer.Args = map[string]string{
+				"HELLO": "world",
+				"FOO":   "bar",
+			}
+			runTest(t, distroSigningTest(t, spec, testConfig.SignTarget))
+		})
+
+		t.Run("with path build arg and build context", func(t *testing.T) {
+			spec := newSpec()
+			spec.PackageConfig.Signer = nil
+
+			signConfig := llb.Scratch().File(llb.Mkfile("/unusual_place.yml", 0o400, []byte(`
+signer:
+  image: `+phonySignerRef+`
+  cmdline: /signer
+`)))
+
+			runTest(t, distroSigningTest(
+				t,
+				spec,
+				testConfig.SignTarget,
+				withBuildContext(ctx, t, "dalec_signing_config", signConfig),
+				withBuildArg("DALEC_SIGNING_CONFIG_CONTEXT_NAME", "dalec_signing_config"),
+				withBuildArg("DALEC_SIGNING_CONFIG_PATH", "/unusual_place.yml"),
+			))
+		})
+
+		t.Run("path build arg takes precedence over spec config", func(t *testing.T) {
+			spec := newSpec()
+			spec.PackageConfig.Signer.Frontend.Image = "notexist"
+
+			signConfig := llb.Scratch().File(llb.Mkfile("/unusual_place.yml", 0o400, []byte(`
+signer:
+  image: `+phonySignerRef+`
+  cmdline: /signer
+`)))
+
+			runTest(t, distroSigningTest(
+				t,
+				spec,
+				testConfig.SignTarget,
+				withBuildContext(ctx, t, "dalec_signing_config", signConfig),
+				withBuildArg("DALEC_SIGNING_CONFIG_CONTEXT_NAME", "dalec_signing_config"),
+				withBuildArg("DALEC_SIGNING_CONFIG_PATH", "/unusual_place.yml"),
+			))
+		})
+
+		t.Run("with path build arg and build context", func(t *testing.T) {
+			spec := newSpec()
+			spec.PackageConfig.Signer = nil
+
+			signConfig := llb.Scratch().File(llb.Mkfile("/unusual_place.yml", 0o400, []byte(`
+signer:
+  image: `+phonySignerRef+`
+  cmdline: /signer
+`)))
+
+			runTest(t, distroSigningTest(
+				t,
+				spec,
+				testConfig.SignTarget,
+				withBuildContext(ctx, t, "dalec_signing_config", signConfig),
+				withBuildArg("DALEC_SIGNING_CONFIG_CONTEXT_NAME", "dalec_signing_config"),
+				withBuildArg("DALEC_SIGNING_CONFIG_PATH", "/unusual_place.yml"),
+			))
+		})
+
+		t.Run("with no build context and config path build arg", func(t *testing.T) {
+			spec := newSpec()
+			spec.PackageConfig.Signer = nil
+
+			signConfig := llb.Scratch().
+				File(llb.Mkdir("/test/fixtures/signer/", 0o755, llb.WithParents(true))).
```

Comment:
Can we trim these paths down since they aren't needed?

Reply:
done